Added log level option to commands.

patriksvensson opened the pull request:
This is a proposal for fixing #120.

I'm not very familiar with your code base, so I apologize if it there are discrepancies in code style or similar. I've tried to keep the code as idiomatic to the one in the code base.

If you have any feedback for changes and/or improvements I will happily change my code. : smile:

## ApiCommand based commands

![image](https://user-images.githubusercontent.com/357872/32246592-bc1f3d4c-be7f-11e7-8e89-cc431ea59230.png)

## Pack command

![image](https://user-images.githubusercontent.com/357872/32246557-9cdbb7bc-be7f-11e7-9ee2-b819c6205943.png)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OctopusDeploy/OctopusClients?pullRequest=201) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/OctopusDeploy/OctopusClients?pullRequest=201) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OctopusDeploy/OctopusClients?pullRequest=201) it.</sub>

mcasperson commented:
Hi Patrik, thanks for this PR. We like the idea, but unfortunately in its current for the new logging options don't work as expected.

In Program.cs, this code:

```
                if (command is ILogLevelAware logLevelAwareCommand)
                {
                    if (logLevelAwareCommand.LogLevel != null)
                    {
                        ConfigureLogger(logLevelAwareCommand.LogLevel);
                    }
                }
```

is executed before this code:

```
command.Execute(args.Skip(1).ToArray()).GetAwaiter().GetResult();
```

`command.Execute` is where the command line params are parsed. So prior to `command.Execute`, `logLevelAwareCommand.LogLevel` is always null. This means the log level is never changed.

patriksvensson commented:
@mcasperson Ok. That’s strange. I will take a look at it but I tried it out and it seemed to work ad expected...

I will take a look and update the PR in a couple hours.

patriksvensson commented:
@mcaspwrson Ok, I see it now. Seems to be an issue of ”10 unit tests, 0 integration tests” :smile:

The fix should be easy. I’ll update te PR in a couple of hours.

patriksvensson commented:
@mcasperson Ok, I've updated the PR. It's not very testable in the current state, but I knowingly opted for a simple solution which didn't require us to inject Serilog's `LoggingLevelSwitch` in every command and we can take the discussion from there.

Another approach we could take is to create a decorator for `ILogger` that contains the logging level switch directly. This would make the log level code more testable, but also increase the complexity + maintainability of the code base.

It's possible I've missed something obvious here, if that's the case, please don't hesitate to correct me :smile:

mcasperson commented on source/Octo/Diagnostics/LogUtilities.cs:
```diff
@@ -0,0 +1,54 @@
+            {
+                return level;
+            }
+            throw new CommandException($"Unrecognized loglevel '{value}'.");
```
We try to provide valid options when displaying an error. So this code would print:

> Unrecognized loglevel 'blah'.

Ideally it should print something like:

> Unrecognized loglevel 'blah'. Valid options are fatal, error, warning, information, debug or verbose.

patriksvensson commented:
@mcasperson I've updated the message per your request and I also did some minor changes to the description of the command to match other commands better.

I've also rebased my branch against master.

mcasperson commented:
Thanks for that Patrik. We've merged this, and the changes will go out with the next release.

patriksvensson commented:
@mcasperson Perfect! Do you have any estimation of when you're going to push the new client?
